src/Patterns/Scd4Pattern.php

```diff
@@ -82,43 +61,33 @@ protected function getTemplateVariables(): array
         ];
     }
 
-    private function getInputColumns(): array
-    {
-        return array_merge($this->getParameters()->getPrimaryKey(), $this->getParameters()->getMonitoredParameters());
-    }
-
     private function getSnapshotPrimaryKeyParts(): array
     {
-        // All snapshot columns are lower
-        return $this->columnsToLower(
+        $columns = $this->getColumnsWithDefinition(
             array_merge($this->getParameters()->getPrimaryKey(), [self::COLUMN_SNAPSHOT_DATE])
         );
-    }
 
-    private function getSnapshotInputColumns(): array
-    {
         // All snapshot columns are lower
-        return $this->columnsToLower($this->getInputColumns());
+        return $this->columnsToLower($columns);
     }
 
-    private function getSnapshotSpecialColumns(): array
+    protected function getSnapshotSpecialColumns(): array
     {
-        $columns[] = self::COLUMN_SNAPSHOT_DATE;
-        $columns[] = self::COLUMN_ACTUAL;
+        $columns[] = [
+            'name' => self::COLUMN_SNAPSHOT_DATE,
```

Comment:
tady ty názvy sloupečků jsou pevně daný... podle Pepy z CF to tak být asi nemá, ale není předmětem tohodle PR... dyžtak se to fixne někde bokem